R/zchunk_LA119.solar.R

```diff
@@ -27,44 +27,113 @@ module_energy_LA119.solar_DISABLED <- function(command, ...) {
 
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
-    Land_type_area_ha <- get_data(all_data, "Land_type_area_ha")
+    Land_type_area_ha <- get_data(all_data, "aglu/LDS/Land_type_area_ha")
     Smith_irradiance_ctry_kwh <- get_data(all_data, "energy/Smith_irradiance_ctry_kwh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # 0. Give binding for variable names used in pipeline
+    Area.bm2.R <- Area.share <- GCAM_region_ID <- dni <- dni.R <- dni.area <- dni.area.R <-
+      dni_avg <- dni_avg_rel <- irradiance <- irradiance.R <- irradiance.area <-
+      irradiance.area.R <- irradiance_avg <- irradiance_avg_rel <- iso <-
+      region_GCAM3 <- value <- year <- NULL
+
     # ===================================================
+    # 2. Perform computations
+    # The irradiance data contained other * region categories which we will have to downscale
+    # Seperate the other * region from the ones with valid country codes that do not need additonal
+    # downscaling
+    Smith_irradiance_ctry_kwh %>%
+      filter(iso %in% grep("OTHER", iso, value = TRUE)) %>%
+      # Rename columns to indicate these are the regional values which will be used later
+      # to downscale to the country
+      mutate(region_GCAM3 = sub("OTHER_", "", iso)) %>%
+      select(-iso) %>%
+      rename(irradiance.R = irradiance,
```

Comment:
I'm concerned that the ".R" used throughout may be confused with the R coding language. Could we use "reg" instead?

Reply:
Good point! It confused me when I started re-writing the chunk. Using `reg` sounds good to me unless `.R` is also used in other chunks that we want to be consistent with. Do we have a decision? @bpbond

Reply:
I think for file names, we need to leave alone. But for local variable names, it should be fine.

Reply:
Agreed--good suggestion @skim301 , thanks. No, I've never seen another ".R" variable in the data system.